### PR TITLE
Miscellaneous minor improvements

### PR DESCRIPTION
**Pull request by bwohlberg**
Miscellaneous minor improvements

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -62,5 +62,5 @@ SCICO is distributed as open-source software under a BSD 3-Clause License (see t
 
 LANL open source approval reference C20091.
 
-(c) 2020-2021. Triad National Security, LLC. All rights reserved.
+(c) 2020-2022. Triad National Security, LLC. All rights reserved.
```
Comment (tbalke) [inline README.rst:65]: While you are at it, `overview.rst` also needs that change.

File: scico/optimize/pgm.py [hunk 2]
```diff
@@ -442,39 +442,41 @@ def x_step(v, L):
...
         if itstat_options:
             default_itstat_options.update(itstat_options)
-        self.itstat_insert_func = default_itstat_options.pop("itstat_func", None)
+        self.itstat_insert_func: Callable = default_itstat_options.pop("itstat_func")  # type: ignore
         self.itstat_object = IterationStats(**default_itstat_options)
 
         self.x: Union[JaxArray, BlockArray] = ensure_on_device(x0)  # current estimate of solution
 
-    def objective(self, x) -> float:
+    def objective(self, x=None) -> float:
```
Comment (tbalke) [inline scico/optimize/pgm.py:476]: I believe previously there was an effort to make these methods purely functional.

@crstngc can you comment on this?

**Review by tbalke — APPROVED** [review]
Approve upon addressing the two minor comments.

File: README.rst (hunk 1 above)
Comment (tbalke) [reply README.rst:65]: Similarly,

`scico/_autograd.py`

`scico/diagnostics.py`

`scico/plot.py`

`scico/linop/_matrix.py`

`scico/linop/_stack.py`

File: scico/optimize/pgm.py (hunk 2 above)
Comment (bwohlberg) [reply scico/optimize/pgm.py:476]: I think that effort was abandoned. I inserted the default argument here for compatibility with all the other optimizers. It was also necessary for the switch to the simplified iterations stats initialization, also part of this PR.

File: README.rst (hunk 1 above)
Comment (bwohlberg) [reply README.rst:65]: Noted on `overview.rst`. The policy on the code is to bump the copyright year the first time a module is edited in a new calendar year. Have any of these been touched this year?

Comment (tbalke) [reply README.rst:65]: I see. The other ones are all >= 2 months old.

**Pull request merged**